tempesta_fw/http_parser.c

```diff
@@ -3223,27 +3223,27 @@ __req_parse_x_forwarded_for(TfwHttpMsg *hm, unsigned char *data, size_t len)
 	__FSM_STATE(Req_I_XFF) {
 		/* Eat OWS before the node ID. */
 		if (unlikely(IS_WS(c)))
-			__FSM_I_MOVE(Req_I_XFF);
+			__FSM_I_MOVE_fixup(Req_I_XFF, 1, 0);
 		/*
 		 * Eat IP address or host name.
 		 *
 		 * TODO: parse/validate IP addresses and textual IDs.
 		 * Currently we just validate separate characters, but the
 		 * whole value may be invalid (e.g. "---[_..[[").
```

Comment:
The comment looks scary in context of #1115: we definitely do not want to identify client by garbage like `---[_..[[`. I'm wondering why don't we have the string in `suspicious_x_forwarded_for` test in `t/unit/test_http_parser.c`. So please add the string to the test. Next, we need a verification of IP address if we use it (previously we just forwarded it to a backend w/o any processing), i.e. need to implement the TODO.

Reply:
Now we check the IP address in https://github.com/tempesta-tech/tempesta/pull/1178/files/18cc11ef8c963acf47447f8ffac211125e2f6d6e#diff-0d15717c2e20453f4271befe0024c2d4R2951

Reply:
Ok, probably we can leave the parser code as is for now. But 2 things are required:

1. a test, probably functional, that we do really block a client with malformed `X-Forwarded-For`

2. block the client if we can not translate its `X-Forwarded-For` value to an IP address instead of just silently ignore the input